### ISSUE TITLE
Packages

### ISSUE THREAD

**Issue by prisae** [emsig/libdlf#3]
# Packages

The library contains all the filters as plain text files. However, we want to add CI-scripts (in `/packages/language-name/`) which create deployable packages for different programming languages.

Planned so far:
- [x] Python #6 
- [x] Julia #17 
- [ ] Matlab
- [ ] Fortran
- [ ] C
- [ ] ...?

**Comment by prisae** [comment]
A first version of Python is implemented. However, it might not be the best way yet. What is definitely still missing:
- [x] Testing for the Python package
- [x] conda-forge deployment (only pip at the moment, a problem with setuptools_scm)

**Comment by kerrykey** [comment]
A first version of a Julia package is implemented. A few remaining steps are needed (and beyond my current know-how):
- [x] Github action to build, including [Registrator for adding to Julia's general registry ](https://github.com/JuliaRegistries/Registrator.jl)
- [x] add version number to Project.toml when building in Github action

**Comment by prisae** [comment]
It might be helpful to add some metadata to the packages. E.g., in Python, something like
```
def key_201_2012():
    ...
    return base, j0, j1

key_201_2012.returns = ['base', 'j0', 'j1']
```
This would make it easy to programmatically check what it returns, and if it can be used for a given use case.

E.g.
```
if not 'j0' in chosen_filter.returns:
    raise ValueError('This application requires a j0-filter; please choose another filter')
```
or similar.

Or add dicts, similar to the caching, with
```
dlflib.hankel.j0 = ['wer_201_2018', 'gupt_61_1997', ...]
dlflib.hankel.j1 = ['wer_201_2018', 'gupt_47_1997', ...]
dlflib.hankel.j2 = [??]  # There are j2 filters
```

The issue is that not all filters have all attributes. But then, not all filters are good for a given application. Anyhow, a code must be able to fail gracefully for filters which do not work at all because they have missing values.

Once we start using `libdlf` in real-world applications it might become apparent what more is needed.

**Comment by kerrykey** [comment]
Yeah, I agree that having the metadata like function returns would be useful, especially for external testing and other automated functions.  Otherwise such codes would have to parse a docstring to see know the contents of a returned tuple.

**Comment by prisae** [comment]
In the Python package, the filter functions have now an attribute `values`, which lists the returned values (without the always returned `base`). Example:
```
>>> import libdlf
>>> libdlf.hankel.wer_201_2018.values
['j0', 'j1']
```
